opentelemetry/CHANGELOG.md

```diff
@@ -2,6 +2,26 @@
 
 ## vNext
 
+Added the ability to prevent recursive telemetry generation through new
+context-based suppression mechanisms. This feature helps prevent feedback loops
+and excessive telemetry when OpenTelemetry components perform their own
+operations.
+
+New methods added to `Context`:
+
+- `is_telemetry_suppressed()` - Checks if telemetry is suppressed in this
+  context
+- `with_telemetry_suppressed()` - Creates a new context with telemetry
```

Comment:
You were planning to update the PR to only have two public functions for now, right? https://github.com/open-telemetry/opentelemetry-rust/pull/2821#discussion_r2000016166

Reply:
cool. Thanks.